assets/src/domain/eventEditor/ui/ticketAssignmentsManager/utils.ts

```diff
@@ -70,19 +71,39 @@ export const getMonthsListFromDatetimes = (dates: Array<Datetime>): OptionsType
 		return acc;
 	}, {});
 
+	return yearWiseMonths;
+};
+
+export const getMonthsListFromDatetimes = (dates: Array<Datetime>): OptionsType => {
+	const yearWiseMonths = getYearWiseMonthsFromDates(dates);
+
 	const list = Object.entries(yearWiseMonths).map(([year, months]) => {
 		return {
 			key: year,
 			label: year,
-			options: Object.entries(months).map(([number, name]) => {
+			options: Object.entries(months).map(([monthNumber, monthName]) => {
 				return {
-					key: `${year}:${number}`,
-					label: name,
-					value: `${year}:${number}`,
+					key: `${year}:${monthNumber}`,
+					label: monthName,
+					value: `${year}:${monthNumber}`,
 				};
 			}),
 		};
 	});
 
 	return list;
 };
+
+export const getYearMonthForNextDate = (dates: Array<Datetime>): string => {
+	const yearWiseMonths = getYearWiseMonthsFromDates(activeUpcoming(dates));
+
+	if (!isEmpty(yearWiseMonths)) {
+		const firstYear = Number(Object.keys(yearWiseMonths)[0]);
+		const firstYearMonths = yearWiseMonths[firstYear];
+		const firstMonth = Number(Object.keys(firstYearMonths)[0]);
```

Comment:
or:
```
const [firstYear] = Number(Object.keys(yearWiseMonths));
const [firstMonth] = Number(Object.keys(firstYearMonths));
```
or even better/safer with ramda's `head`:
```
const firstYear = Number(R.head(Object.keys(yearWiseMonths)));
const firstMonth = Number(R.head(Object.keys(firstYearMonths)));
```

Reply:
using `Number` on array will render it useless

Reply:
`head` is returning only first element:
```
R.head(['fi', 'fo', 'fum']); //=> 'fi'
R.head([]); //=> undefined
```